docs/howto/pipeline_testing.md

```diff
@@ -0,0 +1,183 @@
+# HOWTO: Writing pipeline tests for a package
+
+## Introduction
+
+Elastic Packages are comprised of data streams. A pipeline test exercises ingest pipelines defined for a package's data stream.
+
+## Conceptual process
+
+Conceptually, running a pipeline test involves the following steps:
+
+1. Deploy the Elasticsearch instance (part of the Elastic Stack). This step takes time so it should typically be done once as a pre-requisite to running pipeline tests on multiple data streams.
+1. Upload ingest pipelines to be tested.
+1. Use [Simulate API](https://www.elastic.co/guide/en/elasticsearch/reference/master/simulate-pipeline-api.html) to process logs/metrics with the ingest pipeline.
+1. Compare generated results with expected ones.
+
+## Limitations
+
+At the moment pipeline tests have limitations. The main ones are:
+* Results of simulation can be different from real documents. They depend on the input data which may not have all fields defined.
+* There might be integrations which transform data mostly using Beats processors instead of ingest pipelines. In such cases ingest pipelines are rather plain.
+
+## Defining a pipeline test
+
+Packages have a specific folder structure (only relevant parts shown).
+
+```
+<package root>/
+  data_stream/
+    <data stream>/
+      manifest.yml
+  manifest.yml
+```
+
+To define a pipeline test we must define configuration at two levels: the package level and each dataset's level.
+
+### Dataset-level configuration
+
+Next, we must define configuration for each data stream that we want to pipeline test.
```

Comment:
See comment above. With that you can probably remove the heading and sentence below it.

Reply:
Got it, fixed :)